apps/examples/log_dump/log_dump_main.c

```diff
@@ -42,13 +44,29 @@ int log_dump_main(int argc, char *argv[])
 {
 	int ret = 1;
 	char buf[READ_BUFFER_SIZE];
-	int fd = open("/proc/logsave", O_RDONLY);
+	int fd = open("/proc/logsave", O_RDWR);
 
 	if (fd < 0) {
 		printf("Please ensure that procfs is mounted\n");
-		return 0;
+		return -1;
 	}
 
+	/* Log dump starts automatically from boot up.
+	 * To test start, intentionally add to stop. */
+	if (STOP_LOGDUMP_SAVE(fd) < 0)
+	{
+		printf("Fail to stop log dump, errno %d\n", get_errno());
+		goto errout;
+	}
+
+	if (START_LOGDUMP_SAVE(fd) < 0)
+	{
+		printf("Fail to start log dump, errno %d\n", get_errno());
+		goto errout;
+	}
+
+	printf("This Log Should be saved!!!\n");
```

Comment:
How about adding some log to show not saved log?
```
printf("This log should NOT be saved!!!\n");
if (START_LOGDUMP_SAVE(fd) < 0) {
   ...
}
printf("This log should be saved!!!\n");
...
```

Reply:
Modified.